pkg/process/net/resolver/resolver_test.go

```diff
@@ -97,3 +98,233 @@ func TestLocalResolver(t *testing.T) {
+			Ip:   "127.0.0.1",
+			Port: 1250,
+		},
+	}
+
+	tests := []*model.Connection{
+		{
+			Pid: 1,
+			Laddr: &model.Addr{
+				Ip:   "127.0.0.1",
+				Port: 1234,
+			},
+			Raddr: &model.Addr{
+				Ip:   "10.1.1.2",
+				Port: 1234,
+			},
+			IpTranslation: &model.IPTranslation{
+				ReplDstIP:   "10.1.1.1",
+				ReplDstPort: 1234,
+				ReplSrcIP:   "10.1.1.2",
+				ReplSrcPort: 1234,
+			},
+			NetNS: 1,
+		},
+		{
+			Pid:   2,
+			NetNS: 2,
+			Laddr: &model.Addr{
+				Ip:   "10.1.1.2",
+				Port: 1234,
+			},
+			Raddr: &model.Addr{
+				Ip:   "10.1.1.1",
+				Port: 1234,
+			},
+			IpTranslation: &model.IPTranslation{
+				ReplDstIP:   "10.1.1.2",
+				ReplDstPort: 1234,
+				ReplSrcIP:   "127.0.0.1",
+				ReplSrcPort: 1234,
+			},
+		},
+		{
+			Pid:   3,
+			NetNS: 3,
+			Laddr: &model.Addr{
+				Ip:   "127.0.0.1",
+				Port: 1235,
+			},
+			Raddr: unresolvedRaddr["127.0.0.1:1234"],
+		},
+		{
+			Pid:   5,
+			NetNS: 3,
+			Laddr: &model.Addr{
+				Ip:   "127.0.0.1",
+				Port: 1240,
+			},
+			Raddr: &model.Addr{
+				Ip:   "127.0.0.1",
+				Port: 1235,
+			},
+		},
+		{
+			Pid:   5,
+			NetNS: 4,
+			Laddr: &model.Addr{
+				Ip:   "127.0.0.1",
+				Port: 1240,
+			},
+			Raddr: unresolvedRaddr["127.0.0.1:1235"],
+		},
+		{
+			Pid:   10,
+			NetNS: 10,
+			Laddr: unresolvedLaddr["127.0.0.1:1240"],
+			Raddr: unresolvedRaddr["10.1.1.1:1235"],
+		},
+		{
+			Pid:   11,
+			NetNS: 10,
+			Laddr: unresolvedLaddr["127.0.0.1:1250"],
+			Raddr: unresolvedRaddr["127.0.0.1:1240"],
+		},
+		{
+			Pid:   20,
+			NetNS: 20,
+			Laddr: &model.Addr{
+				Ip:          "1.2.3.4",
+				Port:        1234,
+				ContainerId: "baz",
+			},
+			Raddr: &model.Addr{
+				Ip:          "1.2.3.4",
+				Port:        1234,
+				ContainerId: "bar",
+			},
+		},
+		{
+			Pid:   6,
+			NetNS: 7,
+			Laddr: &model.Addr{
+				Ip:   "127.0.0.1",
+				Port: 1260,
+			},
+			Raddr: &model.Addr{
+				Ip:   "127.0.0.1",
+				Port: 1250,
+			},
+		},
+		{
+			Pid:   7,
+			NetNS: 7,
+			Laddr: &model.Addr{
+				Ip:   "127.0.0.1",
+				Port: 1250,
+			},
+			Raddr: &model.Addr{
+				Ip:   "127.0.0.1",
+				Port: 1260,
+			},
+		},
+	}
+
+	resolver := &LocalResolver{}
+	resolver.LoadAddrs(
+		[]*containers.Container{
+			{
+				ID:   "foo1",
+				Pids: []int32{1},
+			},
+			{
+				ID:   "foo2",
+				Pids: []int32{2},
+			},
+			{
+				ID:   "foo3",
+				Pids: []int32{3},
+			},
+			{
+				ID:   "foo4",
+				Pids: []int32{4},
+			},
+			{
+				ID:   "foo5",
+				Pids: []int32{5},
+			},
+			{
+				ID:   "foo6",
+				Pids: []int32{6},
+			},
+			{
+				ID:   "foo7",
+				Pids: []int32{7},
+			},
+			{
+				ID:   "bar",
+				Pids: []int32{20},
+			},
+		},
+	)
+
+	resolvedByRaddr := map[string]string{
+		"10.1.1.2:1234":  "foo2",
+		"10.1.1.1:1234":  "foo1",
+		"127.0.0.1:1235": "foo3",
+		"1.2.3.4:1234":   "bar",
+		"127.0.0.1:1250": "foo7",
+		"127.0.0.1:1260": "foo6",
+	}
+
+	resolver.Resolve(&model.Connections{Conns: tests})
+
+	for _, te := range tests {
```

Comment:
I appreciate the amount of effort in writing these tests and making the assertion messages very clear, but I found it super hard to follow.

Perhaps you could have something like
```go
type testCase struct {
	c *model.Connection
	expectedLaddrID string
	expectedRaddrID string
}
```

and structure each test case like:
```go
testCase{
	c: *model.Connection{
		{
			Pid: 1,
			Laddr: &model.Addr{
				Ip:   "127.0.0.1",
				Port: 1234,
			},
			Raddr: &model.Addr{
				Ip:   "10.1.1.2",
				Port: 1234,
			},
			IpTranslation: &model.IPTranslation{
				ReplDstIP:   "10.1.1.1",
				ReplDstPort: 1234,
				ReplSrcIP:   "10.1.1.2",
				ReplSrcPort: 1234,
			},
			NetNS: 1,
		},
	},
	expectedLaddrID: "foo",
	expectedRaddrID: "bar",
}
```
to make it more readable?

Reply:
Yes, that is a good suggestion. Let me make the changes.